core/app/models/spree/product/scopes.rb

```diff
@@ -197,7 +197,8 @@ def self.available(available_on = nil)
           def self.with_variant_sku_cont(sku)
             sku_match = "%#{sku}%"
             variant_table = Spree::Variant.arel_table
-            subquery = Spree::Variant.where(variant_table[:sku].matches(sku_match).and(variant_table[:product_id].eq(arel_table[:id])))
```

Comment:
Not sure to get why we need the scope to be defined here. This way every call to `with_variant_sku_cont` will give us also discarded results, even if not called with the `Show Deleted` option. Can you please help me understand?

Reply:
when you soft delete a product,  the corresponding variant is deleted(soft delete). This won't be a problem when searching an active SKU but when searching a deleted product, the default select query for the variant won't account for the deleted SKU. And thus we won't get back the deleted products.

Reply:
I think we may add some unit tests here to describe this behavior. What do you think?

Reply:
yes absolutely. I'll get right to it.

Reply:
I have just pushed a commit for tests covering the `with_variant_sku_cont` scope. As for test covering the deletion of the product with their variances, it is covered here:
REF https://github.com/solidusio/solidus/blob/b0092d0f3bda447494a586bf8eb96a18322b8bed/backend/spec/controllers/spree/admin/products_controller_spec.rb#L261